akka-actor/src/main/scala/akka/io/dns/DnsProtocol.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns
+
+import java.util
+
+import akka.annotation.ApiMayChange
+
+import scala.collection.immutable
+import scala.collection.JavaConverters._
+
+/**
+ * Supersedes [[akka.io.Dns]] protocol.
+ *
+ * Note that one MUST configure `akka.io.dns.resolver = async` to make use of this protocol and resolver.
+ *
+ * Allows for more detailed lookups, by specifying which records should be checked,
+ * and responses can more information than plain IP addresses (e.g. ports for SRV records).
+ *
+ */
+@ApiMayChange
+object DnsProtocol {
+
+  def resolve(name: String): Resolve =
+    Resolve(name, Ip())
+
+  def resolve(name: String, requestType: RequestType): Resolve =
+    Resolve(name, requestType)
```

Comment:
Wouldn't we normally rather have additional companion apply-methods for these message factories? Does it really help to have the class internal and the factories public? Also, the package private type anyways escapes it's privacy in the return type here.

Reply:
Yes i think so

Reply:
Agreed